3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py

```diff
@@ -75,30 +74,50 @@
     results.append(result)
     all_GSEA_results.append(result.assign(z_dim=f"z_{col}"))
     all_signatures.append(df)
-    neg_df = df.copy()
-    neg_df[col] = neg_df[col].sample(frac=1).reset_index(drop=True)
+
+
+# In[7]:
+
+
+# Generating negative control model for gsea
+
+neg_signature = signature.iloc[:, 1:].copy()
+
+for col in neg_signature.columns:
+    neg_signature.loc[:, col] = np.random.permutation(neg_signature.loc[:, col].values)
+
+genes = signature.iloc[:,:1]
+neg_signature.insert(0,'0', genes)
+
+neg_GSEA_results = []
+negative_control = []
+
+range = signature.shape[1]
+
+for col in neg_signature.iloc[:,1:range].columns:
+    neg_df = neg_signature.iloc[:,[0,int(col)]]
     neg_result = blitz.gsea(neg_df, library)
     neg_GSEA_results.append(neg_result.assign(z_dim=f"z_{col}"))
     negative_control.append(neg_df)
 
 
-# In[7]:
+# In[8]:
 
 
 # stack up all of the results to be analyzed
 all_GSEA_results= pd.concat(all_GSEA_results)
 neg_GSEA_results = pd.concat(neg_GSEA_results)
 
 
-# In[8]:
+# In[9]:
 
 
 # sort by what you want to evaluate
 all_GSEA_results.sort_values(by='pval', ascending = True)
 #neg_GSEA_results.sort_values(by='pval', ascending = True)
```

Comment:
Consider removing this line of code if not going to be used in the future.

Reply:
Good catch again, I used that line to ensure the control gsea was producing proper results